src/Swarm/App.hs

```diff
@@ -58,17 +65,41 @@ appMain seed scenario toRun cheat = do
           threadDelay 33_333 -- cap maximum framerate at 30 FPS
           writeBChan chan Frame
 
-      -- Run the app.
+      eventHandler <- case port of
+        Nothing -> pure handleEvent
+        Just port' -> do
+          -- Share a reference to the state with the web ui
+          gsRef <- newIORef (s ^. gameState)
+          void $ forkIO $ webMain port' gsRef
+          pure $ \e -> do
+            s' <- get
+            liftIO $ writeIORef gsRef (s' ^. gameState)
+            handleEvent e
```

Comment:
I think it would also make sense to always start the service by default and use this handler.

Reply:
I am not against it, but what will happen if 8080 is already taken?

Reply:
Depending on how important the service is, we could:
- fail to start the game
- pick the next available port (and add the info to the help page or something)
- don't start the api

Or perhaps, we could make the game fail to start only if the user explicitly provide a port number, otherwise we simply continue without the api.

Also we could use a less common port, for example 5357 (which is SW in hex)

Reply:
Failing to start just because the default port is already taken sounds like a terrible option.  Otherwise I don't have a strong opinion.

Reply:
I agree re: always starting the service by default, perhaps with a way to explicitly disable it?

Reply:
I also vote to continue without API, but I was thinking more about discoverability from the player's perspective.
Basically logging anything to stderr or stdout does not make sense - you will only see that after closing the game.

A simple `txt` widget in <kbd>F1</kbd> Help modal would work better IMO, but that one seems buggy atm. We can work on that later though, a (new?) config overview modal could be a lot of work.

Reply:
Alright, I think the last commit and the updated description should resolve this conversation.

Reply:
@xsebek about user notification, in #627 I suggest we re-use the existing message queue.